lib/action_dispatch/session/active_record_store.rb

```diff
@@ -90,29 +96,52 @@ def set_session(env, sid, session_data, options)
             sid
           end
         end
+        alias_method :write_session, :set_session
 
-        def destroy_session(env, session_id, options)
+        def destroy_session(request, session_id, options)
           logger.silence_logger do
-            if sid = current_session_id(env)
-              get_session_model(env, sid).destroy
+            if sid = current_session_id(request)
+              get_session_model(request, sid).destroy
+              env = request.is_a?(ActionDispatch::Request) ? request.env : request
               env[SESSION_RECORD_KEY] = nil
             end
 
             generate_sid unless options[:drop]
           end
         end
+        alias_method :delete_session, :destroy_session
 
-        def get_session_model(env, sid)
-          if env[ENV_SESSION_OPTIONS_KEY][:id].nil?
-            env[SESSION_RECORD_KEY] = find_session(sid)
-          else
-            env[SESSION_RECORD_KEY] ||= find_session(sid)
+        def get_session_model(request, id)
+          logger.silence_logger do
+            model = @@session_class.find_by_session_id(id)
+            if !model
+              id = generate_sid unless ActiveRecord.version.to_s.start_with?('4.')
```

Comment:
What do you mean by "in case the implementation changes"? I added the `#to_s` because `Rails.version` returns a `String` and at the moment I was getting a `Gem::Version` from `AR.version` (`Rails.version` was undefined in this scope). 

Also, the new `Gemspec`only indicates compatibility until prior to 5.1. I doubt they'd change the implementation of this function barring a newly discovered bug or security issues that somehow forced a specific return type that couldn't be cast to a string..

The fact that the `start_with?` arg is suffixed with a dot should avoid ambiguity, I would think, even if it makes the solution hacky. That is, unless you're thinking of some case that I'm not (in which case please tell me).

Regardless though, `ActiveRecord::VERSION::MAJOR == 4` is much cleaner, so assuming it works in all relevant Rails versions, I'll switch the code over to that tonight. Thanks for the suggestion!